src/document.c

```diff
@@ -1520,6 +1520,24 @@ void document_open_files(const GSList *filenames, gboolean readonly, GeanyFilety
 }
 
 
+static void on_keep_edit_history_on_reload_response(GtkWidget *bar, gint response_id, GeanyDocument *doc)
+{
+	if (response_id == GTK_RESPONSE_NO)
+	{
+		file_prefs.keep_edit_history_on_reload = FALSE;
+		document_reload_force(doc, doc->encoding);
```

Comment:
It probably would be great if we could drop the history without actually reloading a second time (`sci_empty_undo_buffer`/`document_undo_clear`, but I'm not 100% sure why/if we don't have a unified way)